common/src/View/CameraTool2D.cpp

```diff
@@ -70,6 +87,29 @@ static void zoom(
 
 void CameraTool2D::mouseScroll(const InputState& inputState)
 {
+  if (pref(Preferences::CameraTrackpadMode))
+  {
+    const float factor = pref(Preferences::CameraMouseWheelInvert) ? -1.0f : 1.0f;
+
+    if (inputState.scrollZ() != 0.0f)
+    {
+      const float zoomFactor = 1.0f + inputState.scrollZ() * factor;
+      const auto mousePos = vm::vec2f{inputState.mouseX(), inputState.mouseY()};
+
+      if (zoomFactor > 0.0f)
+      {
+        zoom(m_camera, mousePos, zoomFactor);
+      }
+    }
+    else
+    {
+      const auto delta = -m_camera.right() * factor * inputState.scrollX()
+                         + m_camera.up() * factor * inputState.scrollY();
+      m_camera.moveBy(delta * moveSpeed(m_camera, false));
+    }
+    return;
```

Comment:
I think I would prefer an `else` over the early return.

Reply:
I did the early `return` mostly to avoid touching the other lines with the indentation change, if that's not an issue I can use `else`.